collections/threads.go

```diff
@@ -0,0 +1,204 @@
+package collections
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/textile/api/common"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	threadNameRx *regexp.Regexp
+
+	ErrInvalidThreadName = fmt.Errorf("name may only contain alphanumeric characters or non-consecutive hyphens, and cannot begin or end with a hyphen")
+)
+
+func init() {
+	threadNameRx = regexp.MustCompile(`^[A-Za-z0-9]+(?:[-][A-Za-z0-9]+)*$`)
+}
+
+type Thread struct {
```

Comment:
This model tracks a thread and who owns it. They can be owned directly by a dev/org, or by a user via an API key.